plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoRow.java

```diff
@@ -0,0 +1,129 @@
+import static java.util.Objects.requireNonNull;
+
+public class TrinoRow
+        implements StructLike
+{
+    private final List<Type> types;
+    private final Block[] blocks;
+    private final int position;
+
+    public TrinoRow(List<Type> types, Block[] blocks, int position)
+    {
+        this.types = requireNonNull(types, "types list is null");
+        this.blocks = requireNonNull(blocks, "blocks array is null");
+        checkArgument(position >= 0, "page position must be positive: %s", position);
+        this.position = position;
+    }
+
+    public int getPosition()
+    {
+        return position;
+    }
+
+    @Override
+    public int size()
+    {
+        return blocks.length;
+    }
+
+    @Override
+    public <T> T get(int i, Class<T> aClass)
+    {
+        Block block = blocks[i].getLoadedBlock();
+        Type type = types.get(i);
+        T value;
+        // TODO: can refactor with IcebergPageSink.getIcebergValue
+        if (block.isNull(position)) {
+            value = null;
+        }
+        else if (type instanceof BigintType) {
+            value = aClass.cast(type.getLong(block, position));
+        }
+        else if (type instanceof IntegerType || type instanceof SmallintType || type instanceof TinyintType || type instanceof DateType) {
+            value = aClass.cast(toIntExact(type.getLong(block, position)));
+        }
+        else if (type instanceof BooleanType) {
+            value = aClass.cast(type.getBoolean(block, position));
+        }
+        else if (type instanceof DecimalType) {
+            value = aClass.cast(readBigDecimal((DecimalType) type, block, position));
+        }
+        else if (type instanceof RealType) {
+            value = aClass.cast(intBitsToFloat(toIntExact(type.getLong(block, position))));
+        }
+        else if (type instanceof DoubleType) {
+            value = aClass.cast(type.getDouble(block, position));
+        }
+        else if (type.equals(TIME_MICROS)) {
+            value = aClass.cast(type.getLong(block, position) / PICOSECONDS_PER_MICROSECOND);
+        }
+        else if (type.equals(TIMESTAMP_MICROS)) {
+            value = aClass.cast(type.getLong(block, position));
+        }
+        else if (type.equals(TIMESTAMP_TZ_MICROS)) {
+            value = aClass.cast(timestampTzToMicros(getTimestampTz(block, position)));
+        }
+        else if (type instanceof VarbinaryType) {
+            value = aClass.cast(type.getSlice(block, position).getBytes());
+        }
+        else if (type instanceof VarcharType) {
+            value = aClass.cast(type.getSlice(block, position).toStringUtf8());
+        }
+        else {
+            // will most likely throw unsupported exception
+            value = block.getObject(position, aClass);
+        }
+
+        return value;
+    }
+
+    @Override
+    public <T> void set(int i, T t)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "writing to TrinoRow is not supported");
```

Comment:
`TrinoException` should be used only when we know what is the reason of the failure.

```
throw new UnsupportedOperationException();
```